### PR TITLE
619 variants bug

### PR DESCRIPTION
**Pull request by dcolligan**

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, thanks. We should add a test to confirm that we have implemented getVariants correctly though. test_simulated_stack is probably the best place for this. Looking at it now, we don't have any ``testGetVariant`` in there, so we should definitely fill this gap in the tests as part of this PR.

**Comment by dcolligan** [comment]
@jeromekelleher updated

File: tests/unit/test_simulated_stack.py [hunk 1]
```diff
@@ -334,6 +334,29 @@ def testGetVariantSet(self):
...
+        responseData = self.sendSearchRequest(
+            path, request, protocol.SearchVariantsResponse)
+        variant = responseData.variants[0]
+
+        # get 'the same' variant using the get method
+        path = '/variants'
+        responseObject = self.sendGetObject(
+            path, variant.id, protocol.Variant)
+        self.assertEqual(responseObject.referenceName, referenceName)
+        self.assertEqual(responseObject.start, start)
```
Comment (jeromekelleher) [inline tests/unit/test_simulated_stack.py]: Why can't we do a deep equality test here? The simulated variant should be exactly the same whichever way we get it, that's why we seed the RNG with the position. We should be able to directly compare the objects, i.e., ``variant1 == variant2``. Also, testing just the first one seems a bit wimpy --- set a page size of 10 (say) and do a GET for all of them?

Comment (dcolligan) [inline tests/unit/test_simulated_stack.py]: The ids don't come back the same, but I guess that means there's an issue with our random number generation... I'll look into it...

**Comment by dcolligan** [comment]
@jeromekelleher fixed.  the issue was that in `getVariants` there was an extra call to the seeded generator before `generateVariant` was called, resulting in a different-state generator getting passed in and misaligning it with what is passed in to `generateVariant` via `getVariant`.

**Comment by jeromekelleher** [comment]
Nice, well spotted. +1.

**Pull request merged**